### DISCUSSION TITLE
[COMMISSIONER JOINER] Issue on JoinerId using OtCommissionerRemoveJoinerWithDiscerner method

### DISCUSSION THREAD

**Discussion by estelleOl** [openthread/openthread#9399] [Q&A] [answered]
Hello, 

I have trouble with the parameter `aJoinerId` of the callback "commissionerJoinerCallback" when I am using the OtCommissionerRemoveJoinerWithDiscerner method.

The `aJoinerId` seem to return NULL pointer when I use the method OtCommissionerRemoveJoinerWithDiscerner. Here is my code:
 
```
/* Add a joiner */
 status = OtCommissionerAddJoinerWithDiscerner(PtOpenThreadInstance, &joinerDiscernerSet, psk, timeoutSec);
        if(status != OT_ERROR_NONE){
          return;
        }
        
 HAL_Delay(1000);

/* Remove the added joiner */
 status = OtCommissionerRemoveJoinerWithDiscerner(PtOpenThreadInstance, &joinerDiscernerSet);
        if(status != OT_ERROR_NONE){
          return;
        }
```

The joiner is well added I can see all parameters I have set. The joiner is also well removed and trig the commissionerJoinerCallback. 
I can check all the parameter is this callback except aJoinerId (which the seems to be NULL).

Here is the callstack I get when I call the OtCommissionerRemoveJoinerWithDiscerner method:

```
commissionerJoinerCallback(...)
void ot::Callback<void (*)(otCommissionerJoinerEvent, otJoinerInfo const *, otExtAddress const *, void *)>::Invoke<otCommissionerJoinerEvent, otJoinerInfo *, ot::Mac::ExtAddress *>(enum otCommissionerJoinerEvent &&, struct otJoinerInfo * &&, class ot::Mac::ExtAddress * &&) const ***
ot::MeshCoP::Commissioner::SignalJoinerEvent(enum ot::MeshCoP::Commissioner::JoinerEvent, struct ot::MeshCoP::Commissioner::Joiner const *) const
ot::MeshCoP::Commissioner::RemoveJoinerEntry(struct ot::MeshCoP::Commissioner::Joiner &)
ot::MeshCoP::Commissioner::RemoveJoiner(class ot::Mac::ExtAddress const *, class ot::MeshCoP::JoinerDiscerner const *, unsigned int)
OtCommissionerRemoveJoinerWithDiscerner(...)
```
With a debugger when I check the value of variable in SignalJoinerEvent  I got :
`aJoiner->mType` equal to `Joiner::kTypeDiscerner` and `aJoiner` not equal to `mActiveJoiner` so the function put noJoinerId to true.
And then invoke the callback commissionerJoinerCallback with a JoinerId to NULL pointer.

Am I doing something wrong or is a case missing for `Joiner::kTypeDiscerner` ?

I hope my explanations are clear,

Regards,

Estelle

**Comment by jwhui** [comment, marked as answer]
What you are observing is expected.

Thread defines how to derive the Joiner ID from the device's IEEE EUI-64. Note that the Joiner Discerner is specific to OpenThread implementation and not a part of the Thread Specification. The callback does not pass the Joiner ID because it does not have the device's IEEE EUI-64.

See implementation:

https://github.com/openthread/openthread/blob/c6bece0729f0c962766ccbf01d2178e984c9f8d4/src/core/meshcop/commissioner.cpp#L115-L127

**Comment by estelleOl** [comment]
Thank you for your response.
